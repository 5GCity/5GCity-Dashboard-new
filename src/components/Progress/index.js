/**
 * Progress Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */

import styled from 'styled-components'
import { Progress } from 'element-react'

export default styled(Progress)`

witdh:100%;

.el-progress-bar__inner {
  background-color: #8CC14E;
  border-radius:0px;
}

${({status}) => status === 'warning' && `
.el-progress-bar__inner {
  background-color: #DDD16C;
  border-radius: 0px;
}
`}

${({status}) => status === 'danger' && `
.el-progress-bar__inner {
  background-color: #DD6C6C;
  border-radius: 0px;
}
`}

.el-progress-bar__outer {
  border-radius: 0px;
  background-color: rgba(137,151,159,0.15);

  height: 8px !important;
}
`
