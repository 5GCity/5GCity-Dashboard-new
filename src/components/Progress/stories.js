/**
 * Progress Component Stories
 * Please write a description or remove this line
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles';

import Progress from './index'

const bStyle = {
  padding:'30px',
  backgroundColor: Theme.bodyBackground,
  height:'100%'
};

const cpus = {
	height: '12px',
	width: '60px',
	color: '#FFFFFF',
	fontFamily: 'Open Sans',
	fontSize: '12px',
	lineHeight: '12px'
}


const example_1 = () => (
  <div style={bStyle}>
    <p style={cpus}>12/64 GB</p>
    <Progress percentage={20} type={'line'} textInside={false} showText={false} /><br/>
    <p style={cpus}>45/64 GB</p>
    <Progress percentage={70} type={'line'} textInside={false} showText={false} status={'warning'}/><br/>
    <p style={cpus}>63/64 GB</p>
    <Progress percentage={99} type={'line'} textInside={false} showText={false} status={'danger'}/>
  </div>
)


storiesOf('Progress', module)
  .add('line', example_1)