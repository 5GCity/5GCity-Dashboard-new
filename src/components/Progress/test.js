/**
 * Progress Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Progress from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Progress />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
