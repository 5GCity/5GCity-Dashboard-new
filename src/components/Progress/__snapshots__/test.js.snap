// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  witdh: 100%;
}

.c0 .el-progress-bar__inner {
  background-color: #8CC14E;
  border-radius: 0px;
}

.c0 .el-progress-bar__outer {
  border-radius: 0px;
  background-color: rgba(137,151,159,0.15);
  height: 8px !important;
}

<div
  className="el-progress el-progress--line c0"
  style={Object {}}
>
  <div
    className="el-progress-bar"
  >
    <div
      className="el-progress-bar__outer"
      style={
        Object {
          "height": "6px",
        }
      }
    >
      <div
        className="el-progress-bar__inner"
        style={
          Object {
            "width": "0%",
          }
        }
      />
    </div>
  </div>
  <div
    className="el-progress__text"
    style={
      Object {
        "fontSize": "14.4px",
      }
    }
  >
    0%
  </div>
</div>
`;
